modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalResponseTests.java

```diff
@@ -84,6 +109,35 @@ public void testSerialization() throws IOException {
         }
     }
 
+    public void testXContentParsing() throws IOException {
+        RankEvalResponse testItem = createRandomResponse();
+        boolean humanReadable = randomBoolean();
+        XContentType xContentType = randomFrom(XContentType.values());
+        BytesReference originalBytes = toShuffledXContent(testItem, xContentType, ToXContent.EMPTY_PARAMS, humanReadable);
```

Comment:
how do we do here with random fields, forward comp etc.?

Reply:
I added inserting random fields, which of course exposed some issues. Also fixed those in the next commit.